### PR TITLE
chore: update URI for cert import

### PR DESCRIPTION
**Pull request by iamhopaul123**
Related #2694. Specifically, instead of grabbing DNS info from env parameters, we now describe the listener rule for it.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/describe/service.go [hunk 1]
```diff
@@ -108,24 +113,35 @@ func newServiceStackDescriber(opt NewServiceConfig, env string) (*serviceStackDe
 
 // Params returns the parameters of the service stack.
 func (d *serviceStackDescriber) Params() (map[string]string, error) {
+	if d.params != nil {
```
Comment (efekarakus) [inline internal/pkg/describe/service.go:116]: Can we think of any scenario where caching these values will be problematic? is it at all possible that a stack update happens in between calls and causes issues or do you think we're safe?

File: internal/pkg/template/templates/environment/partials/environment-manager-role.yml [hunk 2]
```diff
@@ -19,6 +19,17 @@ EnvironmentManagerRole:
       PolicyDocument:
         Version: '2012-10-17'
         Statement:
+{{- if .ImportCertARNs}}
+        - Sid: ACM
```
Comment (efekarakus) [inline internal/pkg/template/templates/environment/partials/environment-manager-role.yml]: nit:
```suggestion
        - Sid: ImportedCertificates
```

**Review by efekarakus — APPROVED** [review]
LGTM, just a question!

File: internal/pkg/describe/service.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/describe/service.go:116]: I was thinking about the same question but couldn't think of a scenario that's problematic for now. Unless we want to use the same struct to get params and then wait and then get it again. Maybe for `status/show` if we want to support "refreshing periodically to get the latest update" like what `--follow` does for `logs`. However, I think we could add a field to disable this caching when we need it. What do you think?

File: internal/pkg/describe/service.go [hunk 3]
```diff
@@ -108,24 +113,35 @@ func newServiceStackDescriber(opt NewServiceConfig, env string) (*serviceStackDe
...
 	if err != nil {
 		return nil, err
 	}
+	d.params = descr.Parameters
 	return descr.Parameters, nil
 }
 
 // Params returns the outputs of the service stack.
 func (d *serviceStackDescriber) Outputs() (map[string]string, error) {
+	if d.outputs != nil {
```
Comment (dannyrandall) [inline internal/pkg/describe/service.go:129]: just to check, these functions with a cache are only called from a single goroutine, right?

File: internal/pkg/describe/uri.go [hunk 4]
```diff
@@ -94,40 +93,46 @@ func (d *LBWebServiceDescriber) URI(envName string) (string, error) {
...
-		dnsName := fmt.Sprintf("%s.%s", d.svc, envOutputs[envOutputSubdomain])
-		uri.DNSNames = []string{dnsName}
-		uri.HTTPS = true
-	}
-	aliases := envParams[stack.EnvParamAliasesKey]
-	if aliases != "" {
-		value := make(map[string][]string)
-		if err := json.Unmarshal([]byte(aliases), &value); err != nil {
-			return albURI{}, err
+	if !ok || isHTTPS != "true" {
```
Comment (dannyrandall) [inline internal/pkg/describe/uri.go]: nit
```suggestion
	if isHTTPS != "true" {
```

File: internal/pkg/template/templates/workloads/partials/cf/env-controller.yml [hunk 5]
```diff
@@ -5,8 +5,10 @@ EnvControllerAction:
   Properties:
     ServiceToken: !GetAtt EnvControllerFunction.Arn
     Workload: !Ref WorkloadName
+{{- if not .UseImportedCerts}}
```
Comment (dannyrandall) [inline internal/pkg/template/templates/workloads/partials/cf/env-controller.yml]: nit
```suggestion
{{- if and (not .UseImportedCerts) (.Aliases)}}
```
plus remove extra `end`

**Review by dannyrandall — APPROVED** [review]

File: internal/pkg/describe/service.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/describe/service.go:129]: yeah

**Pull request merged**